### PR TITLE
Adding German localization

### PR DESCRIPTION
**Pull request by maxrugen**
Adding German localization  #96 🎉

### CODE REVIEW

**Comment by developit** [comment]
Hiya @maxrugen - do you know if this includes the work done in #103?
Cheers!

**Comment by maxrugen** [comment]
Oh! It does not include #103. Wrote everything from scratch yesterday and today since #157 was still open.

**Comment by developit** [comment]
Oh dear! Think there are things we should pull in from that PR?

**Comment by maxrugen** [comment]
Good question. Really depends on whether @bmatzner wants his translation to be used or not, I'm fairly confident mine is just as fine. Pulling in anything from #103 should not be necessary.

**Comment by developit** [comment]
K - I never got a response from him on whether it was ready for merge, so why don't we merge yours and then we can always look into differences post-merge.

**Comment by maxrugen** [comment]
Great! Thanks a lot.
Just looked up his Twitter – he's not very active there too. Not expecting any response.

I'll definitely look into the possibility of using some of his passages instead of mine.

**Comment by bmatzner** [comment]
Hey guys, don't know why #103 was never merged – it has been sitting there since February… Too bad all the work I did was now apparently of no use…

File: content/lang/de/about/browser-support.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+name: Browserunterstützung
+permalink: '/about/browser-support'
+---
+
+# Browserunterstützung
+
+Prect untersützt moderne Browser (Chrome, Firefox, Safari, Edge) und IE9+. Es sollte in IE7 sowie IE8 funktionieren, setzt allerdings einige Polyfiller voraus. Wenn es beabsichtigt ist, Preact zu benutzen und diese älteren Browser zu untersützen, wird empfohlen, einen Polyfiller wie zum Beispiel [es5-shim] oder [babel-polyfill] zu verwenden.
```
Comment (vedam) [inline content/lang/de/about/browser-support.md]: Preact

File: content/lang/de/guide/api-reference.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+# API-Referenzierung
+
+## `Preact.Component`
+
+`Component` ist eine Basisklasse, die normalerweise als Subklasse zum Erstellen von kraftvollen Preact-Komponenten verwendet wird.
+
+### `Component.render(props, state)`
+
+Die `render()`-Funktion ist voraussetzend für alle Komponenten. Sie kann die props und states der Komponente inspizieren und sollte ein Preact-Element oder `null` zurückgeben.
```
Comment (vedam) [inline content/lang/de/guide/api-reference.md]: Die `render()`-Funktion ist Voraussetzung für alle Komponenten

File: content/lang/de/guide/api-reference.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+### Lebenszyklusmethoden
+
+> _**Tipp:** Wenn du eigene HTML5-Elemente verwendert hast: diese ähneln sich mit den `attachedCallback`- und `detachedCallback`-Lebenszyklusmethoden._
+
+Preact ruft die nachfolgenden Lebenszyklusmethoden auf, falls sie für eine Komponente definiert sind:
+
+| Lebenszyklusmethoden        | Wann sie aufgerufen wird                             				 |
+|-----------------------------|--------------------------------------------------------------|
+| `componentWillMount`        | bevor die Komponente an das DOM eingehanden wird					   |
```
Comment (vedam) [inline content/lang/de/guide/api-reference.md]: bevor die Komponente an das DOM eingehanden wird
`eingehanden`?

Wäre das technisch richtig?:
bevor die Komponente in das DOM eingebunden wird

**Comment by maxrugen** [comment]
Whoopsie, thanks. @developit, approving any changes by @vedam should be fine.

File: content/lang/de/guide/api-reference.md (hunk 3 above)
Comment (maxrugen) [inline content/lang/de/guide/api-reference.md]: Ja, wäre es!

File: content/lang/de/guide/differences-to-react.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+- [Stateless Pure Functional Components]  
+    - _Funktionen, die `props` als Argumente empfangen und JSX/VDOM ausgeben_
+- [Contexts]: Support for `context` was added in Preact [3.0].
+    - _Kontext ist eine experimentelle React-Funktion, wurde aber von anderen Bibliotheken adoptiert._
+- [Refs]: Support for function refs was added in Preact in [4.0]. String refs are supported in `preact-compat`.
+    - _Refs bieten einen weg, gerenderte Elemente und Child-Komponenten zu rendern._
+- Virtual DOM Diffing
+    - _Quasi unabdingbar - Preacts Differenzierung ist simpel, aber effektiv und **[extrem](http://developit.github.io/js-repaint-perfs/) [schnell](https://localvoid.github.io/uibench/)**._
+- `h()`, eine mehr generalisierte Version von `React.createElement`
+- _Diese Idee hieß ursprünglich [hyperscript] und war weit über das React-Ökosystem hinaus wertvoll, daher bewirbt Preact den originalen Standart. ([Lies: warum `h()`?](http://jasonformat.com/wtf-is-jsx))_
```
Comment (vedam) [inline content/lang/de/guide/differences-to-react.md]: Standard

File: content/lang/de/guide/extending-component.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    constructor() {
+        super();
+        (this.mixins || []).forEach( m => Object.assign(this, m) );
+    }
+}
+```
+
+---
+
+> **Fußnote:** Man sollte anmerken, dass Vererbung einen in zerbrechliche parent-child-Beziehungen sperren können. Wenn man auf eine Programmieraufgabe, die adequat mit Vererbung gelöst werden kann, gibt es oftmals einen funktionaleren Weg um das gleiche Ziel zu erreichen, das das Erstellen einer solchen Beziehung unnötig macht.
```
Comment (vedam) [inline content/lang/de/guide/extending-component.md]: adäquat

**Comment by maxrugen** [comment]
Could you maybe create an own branch and open up a new PR? That would speed up the whole editing-process a lot

**Comment by vedam** [comment]
Sorry for the noise. I just wanted to read over, and note the minimal typos. I appreciate your hard work on this.

**Pull request merged**